src/classes/Player.js

```diff
@@ -94,13 +96,22 @@ class Player {
     }
 
     try {
-      const stream = ytdl(queue[this.songEntry], {
-        quality: 'highestaudio',
-        highWaterMark: 1 << 25
-      });
+      let stream;
+      const url = queue[this.songEntry]
+      if (queue[this.songEntry].includes('youtube.com')) {
+        stream = ytdl(queue[this.songEntry], {
+          quality: 'highestaudio',
+          highWaterMark: 1 << 25
+        });
+      } else if (queue[this.songEntry].includes('soundcloud.com') && this.soundcloudClientID !== undefined) {
+        stream = await scdl.download(queue[this.songEntry], this.soundcloudClientID)
+      }
       this.dispatcher = await this.connection.play(stream);
 
+      const info = await scdl.getInfo(url, this.soundcloudClientID)
+      this.song = info.title
```

Comment:
If you do implement the `createSoundcloudStream()` method, can you also add a call these methods after updating `this.song`? This makes it so that the bot's presence gets updated with the current track and I'm not sure that your current implementation is doing it.
 
``` js
if (!this.updateDispatcherStatus()) {
  this.updateSongPresence();
}
```